engine/src/test/java/io/camunda/zeebe/engine/state/appliers/EventAppliersTest.java

```diff
@@ -136,4 +141,33 @@ void shouldGetLatestVersionWhenMultipleRegisteredEventAppliersWithDifferentInten
     // then
     Assertions.assertEquals(expectedVersion, actualVersion);
   }
+
+  @Test
+  void shouldRegisterApplierForAllIntents() {
+    // given
+    final var events =
+        Intent.INTENT_CLASSES.stream()
+            .flatMap(c -> Arrays.stream(c.getEnumConstants()))
+            // Heuristic to detect event intents which are generally in past or present tense
+            // instead of the imperative used for commands.
+            .filter(intent -> intent.name().endsWith("ED") || intent.name().endsWith("ING"))
+            // CheckpointIntent is not handled by the engine
+            .filter(intent -> !(intent instanceof CheckpointIntent))
+            // ProcessInstanceResultIntent is only used for client responses and does not appear on
+            // the log
+            .filter(intent -> !(intent instanceof ProcessInstanceResultIntent));
```

Comment:
🔧 Rather than filtering it out here, it might be better to simply add a NOOP applier. Benefits that I see:
- consistent with other engine related intents
- simpler to maintain when we add other intents that don't appear on the log
- gives the option to eventually add a ProcessInstanceResultIntents to the log